osu.Game.Rulesets.Catch/Objects/JuiceStream.cs

```diff
@@ -54,47 +54,52 @@ private void createTicks()
             var tickDistance = Math.Min(TickDistance, length);
             var spanDuration = length / Velocity;
 
-            var minDistanceFromEnd = Velocity * 0.01;
-
             AddNested(new Fruit
             {
                 Samples = Samples,
                 StartTime = StartTime,
                 X = X
             });
 
-            double lastDropletTime = StartTime;
-
             for (int span = 0; span < this.SpanCount(); span++)
             {
                 var spanStartTime = StartTime + span * spanDuration;
                 var reversed = span % 2 == 1;
 
-                for (double d = 0; d <= length; d += tickDistance)
+                for (double d = 0; d < length - 0.01; d += tickDistance)
                 {
-                    var timeProgress = d / length;
-                    var distanceProgress = reversed ? 1 - timeProgress : timeProgress;
+                    var timeProgressStart = d / length;
+                    var timeProgressEnd = Math.Min((d + tickDistance) / length, 1);
 
-                    double time = spanStartTime + timeProgress * spanDuration;
+                    double startTime = spanStartTime + timeProgressStart * spanDuration;
+                    double endTime = spanStartTime + timeProgressEnd * spanDuration;
 
                     if (LegacyLastTickOffset != null)
                     {
                         // If we're the last tick, apply the legacy offset
-                        if (span == this.SpanCount() - 1 && d + tickDistance > length)
-                            time = Math.Max(StartTime + Duration / 2, time - LegacyLastTickOffset.Value);
+                        if (reversed)
+                        {
+                            if (span == this.SpanCount() - 1 && d == 0)
+                                startTime = Math.Min(EndTime, startTime + LegacyLastTickOffset.Value);
+                        } else
+                        {
+                            if (span == this.SpanCount() - 1 && d + tickDistance > length - 0.01)
```

Comment:
`d + tickDistance > length - 0.01` is not the exact opposite of `d < length - 0.01`. I think it should be `>=`.